src/graphApp.test.js

```diff
@@ -0,0 +1,325 @@
+/**
```

Comment:
Really good tests, the only recommendation I would have is fixing the weird indentations and possibly more comments, but those aren't issues, more for convenience.